regions/shapes/tests/test_polygon.py

```diff
@@ -1,52 +1,109 @@
+        self.pixcoord_outside = PixCoord(1, 2)
 
     def test_repr_str(self):
-        reg_repr = ('<PolygonPixelRegion(vertices=PixCoord(x=[3 4 3], '
-                    'y=[3 4 4]))>')
-        assert repr(self.poly) == reg_repr
+        reg_repr = ('<PolygonPixelRegion(vertices=PixCoord(x=[0 2 0], '
+                    'y=[0 0 3]))>')
+        assert repr(self.reg) == reg_repr
 
-        reg_str = ('Region: PolygonPixelRegion\nvertices: PixCoord(x=[3 4 3],'
-                   ' y=[3 4 4])')
-        assert str(self.poly) == reg_str
+        reg_str = ('Region: PolygonPixelRegion\nvertices: PixCoord(x=[0 2 0],'
+                   ' y=[0 0 3])')
+        assert str(self.reg) == reg_str
+
+    def test_contains_scalar(self):
+        assert self.reg.contains(self.pixcoord_inside)
+        assert self.pixcoord_inside in self.reg
+
+        assert not self.reg.contains(self.pixcoord_outside)
+        assert self.pixcoord_outside not in self.reg
+
+    def test_contains_array_1d(self):
+        pixcoord = PixCoord([1, 1], [1, 2])
+        actual = self.reg.contains(pixcoord)
+        expected = [True, False]
+        assert_equal(actual, expected)
 
-    def _test_basic(self):
-        """
-        TODO: implement these tests!
+        with pytest.raises(ValueError) as exc:
+            pixcoord in self.reg
+        assert 'coord must be scalar' in str(exc)
 
-        Just make sure that things don't crash, but no test of numerical accuracy
-        """
+    def test_contains_array_2d(self):
+        pixcoord = PixCoord(
+            [[1, 1, 1], [1, 1, 1]],
+            [[1, 1, 1], [2, 2, 2]],
+        )
+        actual = self.reg.contains(pixcoord)
+        expected = [[True, True, True], [False, False, False]]
+        assert_equal(actual, expected)
 
-        poly1 = PolygonPixelRegion(([10, 20, 20, 10, 10], [20, 20, 10, 10, 20]))
+    def test_bounding_box(self):
+        bbox = self.reg.bounding_box
+        assert bbox == BoundingBox(ixmin=0, ixmax=3, iymin=0, iymax=4)
 
-        poly2 = PolygonPixelRegion(([15, 25, 25, 15, 15], [23, 23, 13, 13, 23]))
+    def test_to_mask(self):
+        # The true area of this polygon is 3
 
-        assert_allclose(poly1.area, 100)
-        assert_allclose(poly2.area, 100)
+        mask = self.reg.to_mask(mode='center', subpixels=1)
+        assert_allclose(np.sum(mask.data), 5)
 
-        assert PixCoord(13, 12) in poly1
-        assert not PixCoord(13, 12) in poly2
+        # Bounding box and output shape is independent of subpixels,
+        # so we only assert on it once here, not in the other cases below
+        assert mask.bbox == BoundingBox(ixmin=0, ixmax=3, iymin=0, iymax=4)
+        assert mask.data.shape == (4, 3)
 
-        coords = PixCoord([13, 8], [15, 15])
-        assert_equal(poly1.contains(coords), [False, True])
+        # This example is with the default: subpixels=5
+        mask = self.reg.to_mask(mode='subpixels')
+        assert_allclose(np.sum(mask.data), 3.48)
```

Comment:
This polygon mask test is failing on travis-ci:
https://circleci.com/gh/astropy/regions/111?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I kinda expected this, because pixels can land on the edge and results jump.
Probably I should just remove this check for `np.sum(mask.data)` and also all the ones below?
(those are still coverered by `pytest-array` tests checking masks I think, so should be OK, no?)